code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,329 @@
+		if("LZ1")
+			spawn_spot = locate(/obj/docking_port/stationary/marine_dropship/lz1) in SSshuttle.stationary	
+		if("LZ2")
+			spawn_spot = locate(/obj/docking_port/stationary/marine_dropship/lz2) in SSshuttle.stationary
+		else
+			return
+	return ..()
+	
+
+/obj/machinery/computer/camera_advanced/remote_fob/CreateEye()
+	if(spawn_spot)
+		eyeobj = new /mob/camera/aiEye/remote/fobdrone(get_turf(spawn_spot))
+		eyeobj.origin = src
+	return
+
+/obj/machinery/computer/camera_advanced/remote_fob/attackby(obj/item/stack/W, mob/user, params)
+	if(istype(W, /obj/item/stack/voucher/sentry))
+		var/useamount = W.amount
+		sentry_remaining += useamount
+		W.use(useamount)
+		to_chat(user, "<span class='notice'>Sentry voucher redeemed.</span>")
+		playsound(src, 'sound/machines/terminal_insert_disc.ogg', 25, FALSE)
+		return
+	if(istype(W, /obj/item/stack/sheet/metal))
+		if(max_metal >= metal_remaining)
+			to_chat(user, "<span class='notice'>Can't insert any more metal.")
+			return
+		var/useamount = min(W.amount, (max_metal-metal_remaining))
+		metal_remaining += useamount
+		W.use(useamount)
+		to_chat(user, "<span class='notice'>Inserted [useamount] metal sheets.")
+		return
+	if(istype(W, /obj/item/stack/sheet/plasteel))
+		if(max_plasteel >= plasteel_remaining)
+			to_chat(user, "<span class='notice'>Can't insert any more plasteel.")
+			return
+		var/useamount = min(W.amount, (max_plasteel-plasteel_remaining))
+		plasteel_remaining += useamount
+		W.use(useamount)
+		to_chat(user, "<span class='notice'>Inserted [useamount] plateel sheets.")
+		return
+	else
+		to_chat(user, "<span class ='notice'>Invalid material type.</span>")
+	return ..()
+
+/obj/machinery/computer/camera_advanced/remote_fob/GrantActions(mob/living/user)
+	if(off_action)
+		off_action.target = user
+		off_action.give_action(user)
+		actions += off_action
+
+	if(metal_cade)
+		metal_cade.target = src
+		metal_cade.give_action(user)
+		actions += metal_cade
+
+	if(plast_cade)
+		plast_cade.target = src
+		plast_cade.give_action(user)
+		actions += plast_cade	
+
+	if(toggle_wiring)
+		toggle_wiring.target = src
+		toggle_wiring.give_action(user)
+		actions += toggle_wiring
+
+	if(sentry)
+		sentry.target = src
+		sentry.give_action(user)
+		actions += sentry
+
+	eyeobj.invisibility = 0
+
+/obj/machinery/computer/camera_advanced/remote_fob/remove_eye_control(mob/living/user)
+	..()
+	eyeobj.invisibility = INVISIBILITY_ABSTRACT
+	eyeobj.eye_initialized = FALSE
+	return PROCESS_KILL
+
+/obj/machinery/computer/camera_advanced/remote_fob/check_eye(mob/living/user)
+	if(!drone_creation_allowed)
+		to_chat(user, "<span class='notice'>Communication with the drone has been disrupted.</span>")
+		user.unset_interaction()
+	return ..()
+
+/////////////////////////////// Placement Actions
+
+/datum/action/innate/remote_fob //Parent stuff
+	action_icon = 'icons/Marine/remotefob.dmi'
+	var/mob/living/C //the mob using the action
```

Comment:
Do try to use more descriptive variable names, such as `user`, `remote_builder` or whatever.